### PR TITLE
Fix issue with deserializing card that has cycle in linksTo

### PR DESCRIPTION
**Pull request by habdelra**
This issue came down to not handling relative URL’s correctly in our identity context, as well as not detecting cycles in our unsubscribe logic. 

Also, pulling on this loose thread resulted in a bunch of related fixes and cleanup.

### CODE REVIEW

**Comment by habdelra** [comment]
argh, the CI test failure revealed deeper changes that need to be made... working on it...

File: packages/host/app/services/card-service.ts [hunk 1]
```diff
@@ -262,12 +254,19 @@ export default class CardService extends Service {
...
     let initialDoc = await this.serializeCard(card);
     let updatedCard = await api.updateFromSerialized<typeof CardDef>(card, doc);
+    let linkedCards = await this.loadPatchedCards(patchData, new URL(card.id));
+    for (let [field, value] of Object.entries(linkedCards)) {
+      // TODO this triggers a save which is not ideal. perhaps instead we could
+      // introduce a new option to updateFromSerialized to accept a list of
+      // fields to pre-load? which in this case would be any relationships that
+      // were patched in
+      (updatedCard as any)[field] = value;
+    }
```
Comment (habdelra) [inline packages/host/app/services/card-service.ts:269]: the previous logic for patching a card was actually relying on a bug to load linked cards that were added via the patch. now we do that explicitly

File: packages/host/app/services/card-service.ts [hunk 2]
```diff
@@ -280,6 +279,34 @@ export default class CardService extends Service {
...
+        }
+      }),
+    );
+    return result;
+  }
+
+  // TODO let's use better types for cardData and patchData here
   private isPatchApplied(
     cardData: Record<string, any> | undefined,
     patchData: Record<string, any>,
```
Comment (habdelra) [inline packages/host/app/services/card-service.ts:312]: when I used the new `PatchData` type here for the `patchData` there were type errors that would indicate that `isPatchApplied()` doesn't always work.

File: packages/host/app/services/card-service.ts
Comment (habdelra) [inline packages/host/app/services/card-service.ts:1]: i performed a public service and removed all the optional `loader` args to the methods here. we should only ever be using the loaderService.loader's loader in this module.

**Review by backspace — APPROVED** [review]

**Pull request merged**